DbgProvider/Debugger.psm1

```diff
@@ -961,6 +961,9 @@ function ~x
                 {
                     Get-DbgUModeThreadInfo -Current
                 }
+                elseif($threadId -match '~\[\s*([0-9a-f]{4})\s*\]') {
+                    Switch-DbgUModeThreadInfo -Thread ([System.Convert]::ToInt32($matches[1], 16))
```

Comment:
>Switch-DbgUModeThreadInfo [](start = 20, length = 25)

A few things to tweak here:

1. In this ScriptBlock (which starts on line 935), we only want to retrieve the thread--so instead of `Switch-DbgUModeThreadInfo`, we should be calling `Get-DbgUModeThreadInfo` here. This is to facilitate things like just "`~~[bf30]`", where the user only wants to display that thread, not switch to it (which would be "`~~[bf30]s`", and the switching (or whatever the "command" is, if any) will be handled in the next ScriptBlock in the pipeline (which starts around line 976)).

2. The thread ID used with the `~~` syntax is the "system ID", as opposed to the "debugger ID". So the commands "`~3s`" and "`~~[bf30]`" might be equivalent--the first is using the thread's debugger ID; the second is using the thread's system ID.

3. We want to use `TryParse`--we still want to blow up if the thread ID is garbage, but we want to be able to do it with an error that will be more understandable to the user ("Invalid thread id" versus "could not convert value to int").

So this case should look pretty similar to the one below, where we are using `Get-DbgUModeThreadInfo`, but instead of using `-DebuggerId`, we should use `-SystemId` here.

Make sense?

Oh, and one minor nit: personally I wouldn't bother making the regex require the user to give all four digits (the `{4}`)--if there were a thread with ID `0f30`, I think we should just accept `f30` and be happy.

Reply:
I agree with all of your 4 points (1-3 + regex{4}).

As I only have invested a couple of hours trying to understand the flow from inputting a command until it is executed, so I was unaware of the difference between of `Switch-DbgUModeThreadInfo` and `Get-DbgUModeThreadInfo` (nor did I know there was one.)